app/services/UnitLinksEditService.scala

```diff
@@ -7,31 +7,36 @@ import repository.AdminDataUnitLinksEditRepository
 import repository.rest._
 import uk.gov.ons.sbr.models._
 import uk.gov.ons.sbr.models.Period
-import unitref.VatUnitRef
+import uk.gov.ons.sbr.models.edit.AdminDataParentLeuEdit
+import unitref.{ PayeUnitRef, VatUnitRef }
 
 import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 
 class UnitLinksEditService @Inject() (repository: AdminDataUnitLinksEditRepository) extends EditService with LazyLogging {
 
-  override def editVatParentUnitLink(period: Period, vatref: VatRef, editParentLink: EditParentLink): Future[EditParentLinkStatus] = {
+  override def editAdminDataParentUnitLink(period: Period, id: AdminDataParentLeuEdit, editParentLink: EditParentLink): Future[EditParentLinkStatus] = {
     val parent = editParentLink.parent
-    val vatIdAndType = (IdAndType.apply _).tupled(VatUnitRef.toIdTypePair(vatref))
-    val updateParentUnitKey = UnitKey(vatIdAndType.id, vatIdAndType.`type`, period)
+    val (unitId, unitType) = id match {
+      case payeref: PayeRef => PayeUnitRef.toIdTypePair(payeref)
+      case vatref: VatRef => VatUnitRef.toIdTypePair(vatref)
+    }
+    val idAndType = (IdAndType.apply _).tupled(unitId, unitType)
```

Comment:
There is a compiler warning here as a tuple of arguments is not being supplied - and the arguments are being automatically adapted.

Is this not now ```IdAndType(unitId, unitType)``` ?